optics-th/src/Optics/TH/Internal/Product.hs

```diff
@@ -145,21 +145,23 @@ makeFieldLabel
 makeFieldLabel rules (defName, (defType, cons)) = do
   (context, instHead) <- case defType of
     OpticSa _ _ otype s a -> do
+      (k,  cxtK) <- eqSubst (ConT $ opticTypeToTag otype) "k"
```

Comment:
Could you write a description what kind of instance this code generates. I *guess* you make `k ~ A_Lens` stuff, but I don't like guessing...

Or you could take the above as a hint to update `Optics.Label` documentation. `metametapost` also have two instances written out by hand, which could be updated to show "best practices".